src/pages/posts/[postid].astro

```diff
@@ -41,10 +42,22 @@ const html = sanitizeHtml(await marked.parse(content.body), {
 const showCopyLinkWarning = Astro.url.searchParams.get('created') === content.edit_key;
 const url = Astro.url.href.split('?')[0];
 const editURL = url + `/edit?key=${content.edit_key}`;
----
 
+const canViewAdultContent = Astro.url.searchParams.get('adult');
+const showAdultWarning = !canViewAdultContent && content.adult_content;
+---
 <BaseLayout title={`${content.title || 'Untitled'}`}>
     <Seo title={content.title} description={trimmedText} />
+
+    <Modal client:only="preact" isOpen={true} hidePage={true}>
```

Comment:
currently the modal shows for every post, need to check for showAdultWarning

Reply:
Whoops